libs/unity/library/Tests/Runtime/MediaLineTests.cs

```diff
@@ -235,5 +233,66 @@ public IEnumerator SetReceiver()
             // Terminate the coroutine.
             yield return null;
         }
+
+        [UnityTest]
+        public IEnumerator DestroyPeerConnection()
+        {
+            // Create the component
+            var go = new GameObject("test_go");
+            go.SetActive(false); // prevent auto-activation of components
+            var pc = go.AddComponent<PeerConnection>();
+
+            // Add a media line while inactive.
+            VideoTrackSource source1 = go.AddComponent<UniformColorVideoSource>();
+            VideoReceiver receiver1 = go.AddComponent<VideoReceiver>();
+            MediaLine ml1 = pc.AddMediaLine(MediaKind.Video);
+            ml1.Source = source1;
+            ml1.Receiver = receiver1;
+
+            // Media lines have not been set yet.
+            Assert.IsEmpty(source1.MediaLines);
```

Comment:
That's quite unintuitive. So I do `AddMediaLine()` and sometimes it adds it to `MediaLines` and sometimes not?

Reply:
TBC `AddMediaLine` always adds a media line to `PeerConnection.MediaLines`. I assume you are talking about `mediaLine.Source = source`.

The problem is that:
- setting a connection in the editor won't call `OnAddedToMediaLine`. I am not sure if there is a reliable way to do it. This implies that there is inevitably going to be a state (fresh out of load, not awake) where the object is accessible but `mediaLine` is not in `source.MediaLines`
- we can't serialize the connection since `MediaLine` is not a Unity object and therefore Unity won't let you share a reference to the same `MediaLine` across two objects

`OnAfterDeserialize` seemed to suit our case except for the problems above. So is seems some inconsistency in state is inevitable. `Awake` seems to be the next best option. Any better ideas?

Reply:
Ok I see the problem, I have no better solution then.